optimum/intel/ipex/modeling_base.py

```diff
@@ -92,9 +95,7 @@ def ipex_jit_trace(model, task, use_cache):
 
     model.config.return_dict = False
 
-    if "past_key_values" in sample_inputs and use_cache:
-        # Make sure the model will output past_key_values in generation tasks
-        model.config.use_cache = True
+    model.config.use_cache = "past_key_values" in sample_inputs
```

Comment:
also in my previous suggestion I missed that this should be done only for generation models so I think we can do something like :
```suggestion
    if "past_key_values" in sample_inputs:
        model.config.use_cache = True
```
what do you think ?